prime-angular-frontend/src/app/modules/site-registration/pages/organization-claim-page/organization-claim-page.component.ts

```diff
@@ -0,0 +1,112 @@
+import { MatSlideToggleChange } from '@angular/material/slide-toggle';
+import { MatDialog } from '@angular/material/dialog';
+import { Validators } from '@angular/forms';
+
+import { FormUtilsService } from '@core/services/form-utils.service';
+import { OrganizationResource } from '@core/resources/organization-resource.service';
+import { ToastService } from '@core/services/toast.service';
+
+import { Observable, of, Subscription } from 'rxjs';
+import { catchError, exhaustMap, map } from 'rxjs/operators';
+
+import { RouteUtils } from '@lib/utils/route-utils.class';
+import { AbstractEnrolmentPage } from '@lib/classes/abstract-enrolment-page.class';
+import { Party } from '@lib/models/party.model';
+
+import { AuthService } from '@auth/shared/services/auth.service';
+import { BcscUser } from '@auth/shared/models/bcsc-user.model';
+import { SiteRoutes } from '@registration/site-registration.routes';
+import { OrganizationFormStateService } from '@registration/shared/services/organization-form-state.service';
+import { Organization } from '@registration/shared/models/organization.model';
+import { OrganizationClaimPageFormState } from './organization-claim-page-form-state.class';
+import { OrgnizationType } from '@registration/shared/models/organization-type-model';
+
+@Component({
+  selector: 'app-organization-claim-page',
+  templateUrl: './organization-claim-page.component.html',
+  styleUrls: ['./organization-claim-page.component.scss']
+})
+export class OrganizationClaimPageComponent extends AbstractEnrolmentPage implements OnInit {
+  public formState: OrganizationClaimPageFormState;
+  public title: string;
+  public routeUtils: RouteUtils;
+  public isCompleted: boolean;
+  public isClaimExistingOrg: boolean;
+  public hasOrgClaimError: boolean;
+  public busy: Subscription;
+
+  constructor(
+    protected dialog: MatDialog,
+    protected formUtilsService: FormUtilsService,
+    private organizationFormStateService: OrganizationFormStateService,
+    private organizationResource: OrganizationResource,
+    private toastService: ToastService,
+    private authService: AuthService,
+    private route: ActivatedRoute,
+    router: Router
+  ) {
+    super(dialog, formUtilsService);
+
+    this.title = this.route.snapshot.data.title;
+    this.routeUtils = new RouteUtils(route, router, SiteRoutes.MODULE_PATH);
+  }
+
+  public onClaimOrgChange(event: MatSlideToggleChange): void {
+    this.isClaimExistingOrg = event.checked;
+    this.toggleClaimFormValidators(event.checked);
+  }
+
+  public ngOnInit(): void {
+    this.createFormInstance();
+    this.isClaimExistingOrg = !!this.formState.json.pec && !!this.formState.json.claimDetail;
+    this.toggleClaimFormValidators(this.isClaimExistingOrg);
+  }
+
+  protected createFormInstance(): void {
+    this.formState = this.organizationFormStateService.organizationClaimPageFormState;
+  }
+
+  protected patchForm(): void {
+  }
+
+  protected performSubmission(): Observable<boolean> {
```

Comment:
I'm getting this error on submission of a claim for an approved site on the PR.  A site with a PEC of 12345 was submitted by PRIMET003 and has been approved, but PRIMET004 (other test card) can't claim the organization:

![image](https://user-images.githubusercontent.com/599498/127079559-ecb8bfb7-af4f-4e75-ad94-343fdbd82c2e.png)

Reply:
As discussed on Zoom, the (data) problem is that there are 3 sites, each belonging to a separate organization, that have the same Site Id "12345"

![image](https://user-images.githubusercontent.com/7586665/127243208-f2d3abca-4340-49f3-81fd-aa6574fac184.png)

Reply:
https://bcgovmoh.atlassian.net/browse/PRIME-1762